jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/Request.java

```diff
@@ -151,10 +153,27 @@ public interface Request extends Attributes, Content.Source
     Context getContext();
 
     /**
-     * TODO see discussion in #7713, as this path should probably be canonically encoded - ie everything but %25 and %2F decoded
-     * @return The part of the decoded path of the URI after any context path prefix has been removed.
+     * <p>Get the context pat. this is equivalent to <code>request.getContext().getContextPath()</code>
+     *
+     * @param request The request to get the context path from.
+     * @return The contextPath of the request. .
+     * @see Context#getContextPath()
+     */
+    static String getContextPath(Request request)
+    {
+        return request.getContext().getContextPath();
+    }
+
+    /**
+     * <p>Get the canonically encoded path of the URI, scoped to the current context.</p>
+     * <p>The <code>pathInContext</code> represents the targeted resource within the current content for the request.</p>
```

Comment:
I would remove this line (confusing) and replace it with an example instead.

```suggestion
     * <p>For example, when {@code contextPath=/ctx} and the request {@code HttpURI.canonicalPath=/ctx/foo}, then {@code pathInContext=/foo}.</p>
```

I would also mention that it does not include the query string, nor any path parameter.